frontend/Auth/FinishResetPassword.tsx

```diff
@@ -56,60 +60,102 @@ export function FinishResetPassword(props: RouterProps) {
-              type="password"
-              value={finishResetPasswordFormData.password}
-              onChange={(evt) =>
-                setFinishResetPasswordFormData({
-                  ...finishResetPasswordFormData,
-                  password: evt.target.value,
-                })
-              }
-              required
-            />
-          </FormField>
-        </div>
+      {tokenValidationResponse?.tokenIsValid == true &&
+      tokenValidationResponse?.tokenIsExpired == false ? (
+        <form
+          onSubmit={unary(submitMutation.mutate)}
+          className="space-y-32 relative lg:max-w-sm w-64"
+        >
+          <div>
+            <h1 className="text-gray-500 place-self-start mb-2 text-5xl lg:text-xl">
+              Set New Password
+            </h1>
+
+            <p className="text-left text-gray-600 py-8 text-4xl lg:text-sm">
+              Enter your new password below.
+            </p>
+
+            <FormField className="mb-4">
+              <FormFieldLabel
+                className="text-3xl lg:text-xs"
+                htmlFor="username"
+              >
+                Email
+              </FormFieldLabel>
+              <Input
+                id="username"
+                type="email"
+                placeholder={userEmail}
+                value={userEmail}
+                disabled
+                className="text-gray-500"
+              />
+            </FormField>
+
+            <FormField className="mb-4">
+              <FormFieldLabel
+                className="text-3xl lg:text-xs"
+                htmlFor="password"
+              >
+                Password
+              </FormFieldLabel>
+              <Input
+                id="password"
+                className="text-5xl lg:text-sm"
+                type="password"
+                value={finishResetPasswordFormData.password}
+                onChange={(evt) =>
+                  setFinishResetPasswordFormData({
+                    ...finishResetPasswordFormData,
+                    password: evt.target.value,
+                  })
+                }
+                required
+              />
+            </FormField>
+          </div>
+
+          <div className="inset-x-0 my-8 bottom-0">
+            <Button
+              kind={ButtonKind.primary}
+              type="submit"
+              className="w-full h-24 lg:h-10 text-3xl lg:text-sm"
+            >
+              Submit New Password
+            </Button>
+          </div>
+        </form>
+      ) : (
+        <div className="space-y-32 relative lg:max-w-sm w-64">
+          <div>
+            <h1 className="text-gray-500 place-self-start mb-2 text-5xl lg:text-xl">
+              Set New Password
+            </h1>
+
+            {tokenValidationResponse?.tokenIsValid == false ? (
+              <p className="text-left text-gray-600 py-8 text-4xl lg:text-sm">
+                This reset password link is not valid. Please try resetting your
```

Comment:
Perhaps let's make `"resetting your password"` a `<Link>` so that they can click on it? They might not know how to get back to the reset password page

Reply:
Yes! This is a great idea.